### ISSUE TITLE
数据集

### ISSUE THREAD

**Issue by heywaygo3321** [shuiguoc/AGDR-MIL#2]
您好，请问咱们数据集如何分割的呢，种子数是多少呢。

**Comment by shuiguoc** [comment]
Thank you for your interest in our work.
Regarding the dataset splits: For Camelyon16, we used 5‑fold cross‑validation. For the TCGA Lung dataset, we randomly divided it five times with a fixed ratio of training set : validation set : test set = 65 : 10 : 25.
Full details are provided in Section 4.1 (“Datasets and evaluation metrics”) of our paper, where we state: _“Camelyon16 is a dataset focusing on breast cancer detection, collected from the Utrecht University Medical Center (UMCU) and the Radboud University Medical Center (RUMC). We adopted the cross‑validation method which is used in many studies. Specifically, we used the 5‑fold cross‑validation.  The TCGA Lung dataset includes two lung cancer subtypes: 541 lung adenocarcinoma (LUAD) and 512 lung squamous cell carcinoma (LUSC). Referring to the division ratio of previous studies, we divided it five times according to the ratio of training set : validation set : test set = 65 : 10 : 25.”
Additionally, the random seed used throughout our study is 2000, which matches the seed provided in the accompanying code.
I hope this answers your question – please let us know if you need any further clarification.

**Comment by heywaygo3321** [comment]
感谢您的回复，是和[MHIM-MIL](https://github.com/DearCaat/MHIM-MIL)这篇文章相同的方式处理的Camelyon16嘛

**Comment by shuiguoc** [comment]
Thank you for your follow‑up question. You asked whether our processing of the Camelyon16 dataset is the same as that used in the MHIM‑MIL paper. This is a very important point, and I will provide a detailed answer.

Both approaches adopt cross‑validation, meaning we only have training and test sets – in our strategy, we use the test set as the validation set as well (so the test and validation sets are identical). However, there are some differences:
MHIM‑MIL performs 3‑times 3‑fold cross‑validation (as stated in their Section 4.1: “we adopt 3‑times 3‑fold cross‑validation on this dataset to ensure that each slide is used in training and testing”).
Our method (AGRD‑MIL) performs a single 5‑fold cross‑validation.

Thus, we differ in two main aspects:
Split ratio: MHIM‑MIL uses a training‑to‑test ratio of 2:1, whereas we use 4:1.
Number of averaged results: MHIM‑MIL averages over 9 runs (3 times × 3 folds), while we average over 5 runs (5 folds).

I hope this clarifies the differences. Please let me know if you need any further clarification.

**Comment by heywaygo3321** [comment]
感谢您的回复，请问是否能提供原始数据顺序表吗，我使用我处理的[c16]数据集复现论文结果五折交叉验证在AUC92.17，AUC最高96最低88，我认为可能是数据初始排列顺序不同导致的。

**Comment by shuiguoc** [comment]
Thank you for your reply. Regarding your request for the original data order table, I have now updated the dataset split files used for the Camelyon16 experiments (they are located in the dataset/c16 folder).

Concerning the AUC value you obtained—92.17% on the five‑fold cross‑validation, with a maximum of 96% and a minimum of 88%—I believe this variation may also be related to the preprocessing of the whole‑slide images (WSIs). Factors such as the number of patches extracted from each WSI and the threshold used for background removal in CLAM can have a impact on the results. Therefore, as long as your performance does not deviate substantially from the reported range, I would suggest that you do not worry too much and feel free to report your experimental findings with confidence. Your validation, along with appropriate citations, can also serve as a useful reference for future researchers.

I hope this response is helpful to you.

**Comment by heywaygo3321** [comment]
感谢您的开源

**Comment by heywaygo3321** [comment]
请问这是最优参数吗，我跑的结果为“Accuracy = 88.98% ± 4.07%
AUC = 92.24% ± 3.68%
Recall = 89.20% ± 4.16%
Precision = 88.98% ± 4.07%
F1-score = 88.82% ± 4.11%’与原文感觉差距还挺大的，请问我的实验配置存在问题吗？
Model
    in_dim: 1024
    act: relu
    dropout: 0.25
    h_thr: 0.95
    l_thr: 0.5
    mask: 0.2
    mil_1: attention
    mil_2: attention
    lambda: 0.3

Optimizer:
    type: adam
    lr: 0.0002
    weight_decay: 0.00001

Loss:
    type: ce

Earlystop:
    use: True
    monitor: val_loss
    patience: 30
    stop_epoch: 20
    save_best_model_stage: 0

**Comment by shuiguoc** [comment]
Please note that these are indeed the optimal parameters we identified, and your experimental setup is not problematic.
